webapp/channels/src/packages/mattermost-redux/src/actions/content_flagging.ts

```diff
@@ -53,3 +55,65 @@ export function getContentFlaggingConfig(): ActionFuncAsync<ContentFlaggingConfi
         return {data: response};
     };
 }
+
+export function getPostContentFlaggingFields(): ActionFuncAsync<NameMappedPropertyFields> {
+    return async (dispatch, getState) => {
+        let data;
+        try {
+            data = await Client4.getPostContentFlaggingFields();
+        } catch (error) {
+            forceLogoutIfNecessary(error, dispatch, getState);
+            dispatch(logError(error));
+            return {error};
+        }
+
+        dispatch({
+            type: ContentFlaggingTypes.RECEIVED_POST_CONTENT_FLAGGING_FIELDS,
+            data,
+        });
+
+        return {data};
+    };
+}
+
+export function loadPostContentFlaggingFields(): ActionFuncAsync<NameMappedPropertyFields> {
+    // Use data loader and fetch data to manage multiple, simultaneous dispatches
+    return async (dispatch, getState, {loaders}: any) => {
+        if (!loaders.postContentFlaggingFieldsLoader) {
+            loaders.postContentFlaggingFieldsLoader = new DelayedDataLoader<Array<PropertyValue<unknown>>>({
+                fetchBatch: () => dispatch(getPostContentFlaggingFields()),
```

Comment:
question: if is returning the _fields_ and not the _values_, shouldn't the type be `new DelayedDataLoader<NameMappedPropertyFields>`?